scanner/src/funTest/kotlin/storages/ClearlyDefinedStorageFunTest.kt

```diff
@@ -54,7 +54,7 @@ class ClearlyDefinedStorageFunTest : StringSpec({
                     provenance = RepositoryProvenance(
                         vcsInfo = VcsInfo(
                             type = VcsType.GIT,
-                            url = "https://github.com/vdurmont/semver4j/tree/88912638db3f6112a2b345f1638ced33a0a606e1",
+                            url = "https://github.com/vdurmont/semver4j.git",
```

Comment:
Commit: Does clearly defined always return such URLs or only in some cases? ...in case of the latter, should we file an upstream issue?

Reply:
Would be nice to have some reference if available...

Reply:
Sorry, one more question: Previously, revision and resolved revision were guaranteed to be identical by this code. By the new code that guarantee is removed. Should we recover that by adjusting the new code slightly?

Reply:
I don't believe restoring that guarantee is desirable; I rather regard it as a bug that this happened to be the case. And it does not matter in the context of scan storages, as provenance is always matched by resolved revision. So I prefer so allow revision and resolved revision to be different strings, as that can provide additional information about a symbolic name for the resolved revision which might be helpful.

Reply:
But isn't that additional information irrelevant? Because, for clearly defined it's undefined if this additional information has been created when the scan was executed or whether it's solely based on the request.
In the former case it does not matter, in the latter that information is present in the analyzer result.

Reply:
> But isn't that additional information irrelevant?

It depends. It could be interesting during debugging to better understand if something strange happened, like a tag being recreated to point at a different commit.

Anyway, that's hypothetical, but generally I believe having more information at hand is better than having less information.

Reply:
Ok, makes sense. Only the resolved revision get's propagated by the scanner anyway. So the revision remains inside the scanner. See https://github.com/oss-review-toolkit/ort/blob/0d4b1f75932d3a6666d599dbf6359f7e8fcebdff/scanner/src/main/kotlin/Scanner.kt#L233.